packages/utils/src/fallbackSubgraph.ts

```diff
@@ -36,85 +48,102 @@ type Subgraph<T extends SubgraphSdk> = {
+  [SubgraphDomain.TEST]: undefined,
+};
+
+export const graphQuery = async (url: string, query: string): Promise<any> => {
+  return await graphQLRequest(url, query);
+};
+
 /**
  * @classdesc A class that manages the sync status of multiple subgraphs as well as their corresponding SDKs.
  */
-export class FallbackSubgraph<T extends SubgraphSdk> {
+export class FallbackSubgraph<T> {
   // Target maximum calls per second. Subgraphs can be called more than this per second, but it's
   // considered not preferable.
   private static MAX_CPS = 10;
   // Target number of samples in the call metrics for each subgraph call.
   private static readonly METRIC_WINDOW = 100;
 
-  private readonly subgraphs: Subgraph<T>[];
+  private readonly subgraphs: Map<string, Subgraph<T>> = new Map();
 
+  // Syncing queue used to serialize sync calls and for awaiting initial sync to finish when requests
+  // are first made.
+  private readonly syncingQueue: PriorityQueue = new PriorityQueue({ concurrency: 1 });
   private latestSync = 0;
 
   /**
    * Returns boolean representing whether at least one available subgraph is in sync.
    */
   public get inSync(): boolean {
-    return this.subgraphs.some((subgraph) => subgraph.record.synced);
+    return Object.values(this.subgraphs).some((subgraph) => subgraph.record.synced);
   }
 
   /**
    * Returns boolean indicating whether we've synchronized the subgraphs at all (indicating
    * whether the records are in fact representative).
    */
   public get hasSynced(): boolean {
-    return this.subgraphs.some((subgraph) => subgraph.record.syncedBlock !== -1 && subgraph.record.latestBlock !== -1);
+    return Object.values(this.subgraphs).some(
+      (subgraph) => subgraph.record.syncedBlock !== -1 && subgraph.record.latestBlock !== -1,
+    );
   }
 
   public get records(): SubgraphSyncRecord[] {
-    return this.getOrderedSdks().map((sdk) => sdk.record);
+    return this.getOrderedSubgraphs().map((sdk) => sdk.record);
   }
 
   /**
    *
    * @param chainId - Chain ID of the subgraphs.
    * @param sdks - SDK clients along with corresponding URIs used for each subgraph.
    * @param maxLag - Maximum lag value a subgraph can have before it's considered out of sync.
+   * @param domain (default: COMMON) - type of subgraph we are using, whether its the common
+   * @param urls - Preset urls; FallbackSubgraph creates a new empty record for each.
    * @param stallTimeout - the ms we wait until considering a subgraph RPC call to be a timeout.
+   * domain (used for transactions) or the analytics domain.
    */
   constructor(
     private readonly chainId: number,
-    // We use the URIs in sync records for reference in logging.
-    sdks: { client: T; url: string }[],
+    private readonly generateClient: (url: string) => T,
     private readonly maxLag: number,
+    private readonly domain: SubgraphDomain = SubgraphDomain.COMMON,
+    urls: string[] = [],
     private readonly stallTimeout = 10_000,
   ) {
```

Comment:
The interface of this utility class changes a lot, and that's not necessarily the main improvement outlined in the PR body. @rhlsthrm what would the semver procedure for this when none of the other interfaces changed?

Reply:
Consensus: not really guaranteed that we are the only consumers of these functions, will release at `0.2.0`